src/Calculator/StatisticsCalculator.php

```diff
@@ -201,43 +201,25 @@ public function getProductImage($productDetails)
      */
     public function computeConversionByProduct($id_product, $id_product_attribute, $dateStart = null)
     {
-        $queryCarts = new \DbQuery();
-        $queryCarts->select('id_cart');
-        $queryCarts->from('blockwishlist_statistics');
-        $queryCarts->where('id_cart != 0');
-        $queryCarts->where('id_product = ' . $id_product);
-        $queryCarts->where('id_product_attribute = ' . $id_product_attribute);
+        $nbOrderPaidAndShipped = [];
+        $queryOrders = '
+            SELECT count(distinct(o.id_order)) as nb
+            FROM ' . _DB_PREFIX_ . 'orders o
+            LEFT JOIN ' . _DB_PREFIX_ . 'blockwishlist_statistics bws ON (o.id_cart = bws.id_cart )
+            LEFT JOIN ' . _DB_PREFIX_ . 'order_history oh ON (o.`id_order` = oh.`id_order`)
+            LEFT JOIN ' . _DB_PREFIX_ . 'order_state os ON (os.`id_order_state` = oh.`id_order_state` AND os.`paid` = 1 AND os.`shipped` = 1)
+            LEFT JOIN ' . _DB_PREFIX_ . 'order_detail od ON (od.`id_order` = o.`id_order` AND od.`product_id` = ' . (int) $id_product . ' AND od.`product_attribute_id` = ' . (int) $id_product_attribute . ')
```

Comment:
```suggestion
            LEFT JOIN ' . _DB_PREFIX_ . 'order_detail od ON (od.`id_order` = o.`id_order` AND od.`product_id` = bws.`id_product` AND od.`product_attribute_id` = bws.`id_product_attribute`'
```
You already test the product_attribute_id and product_id in the where clause.